src/Dropdown/Dropdown.tsx

```diff
@@ -1,400 +1,160 @@
-  };
-
-  sortOptionsByGroup = (options: SelectNS.Options) => {
-    const { groupByKey } = this.props as DefaultedProps;
-    return sortBy(
-      options,
-      option =>
-        option[groupByKey]
-          ? findIndex(options, o => option[groupByKey] === o[groupByKey])
-          : -1
-    );
+    blurInputOnSelect: true,
+    menuPosition: "bottom",
+    components: {}
   };
 
   getCustomClassNames() {
-    const { size, flat, clearable, menuPosition } = this
-      .props as DefaultedProps;
+    const {
+      size,
+      flat,
+      isClearable,
+      menuPosition,
+      isMulti,
+      isDisabled
+    } = this.props;
     return cx({
       "is-medium": size === "medium",
       "is-small": size === "small",
       "is-flat": flat,
-      "is-clearable": clearable,
+      "is-multi": isMulti,
+      "is-clearable": isClearable,
+      "is-disabled": isDisabled,
       "menu-position-top": menuPosition === "top"
     });
   }
 
-  _onChange = (_value?: Dropdown.Value | null) => {
-    const value = _value === "" || isEmptyArray(_value) ? null : _value;
-    return this.props.onChange(value);
-  };
-
-  onInputKeyDown: SelectNS.OnInputKeyDownHandler = e => {
-    if (e.keyCode === 38 || e.keyCode === 40) {
-      if (isEmptyArray(this.props.options)) {
-        e.preventDefault();
-      }
-    }
-  };
+  private select: Select<OptionType> | null;
 
-  private select: Select | null;
   focus = () => {
     this.select && this.select.focus();
   };
 
-  optionGroupRenderer = (title?: string) => {
-    return title ? (
-      <FlexView className="Select-option-group">
-        {(this.props as DefaultedProps).optionGroupRenderer(title)}
-      </FlexView>
-    ) : null;
-  };
-
-  menuRenderer = (
-    args: SelectNS.MenuRendererProps & Dropdown.SelectMissingMenuRendererProps
-  ) => {
-    const { menuRenderer, groupByKey } = this.props as DefaultedProps;
-    return menuRenderer({
-      ...args,
-      groupByKey,
-      optionGroupRenderer: this.optionGroupRenderer
-    });
-  };
-
   render() {
     const {
-      _onChange,
-      onInputKeyDown,
-      props: { className, options, backspaceRemoves, clearable, ...props }
+      props: {
+        className,
+        backspaceRemovesValue,
+        isClearable,
+        menuPosition,
+        components: customComponents,
+        ...props
+      }
     } = this;
 
-    const selectProps = {
-      ...props,
-      options: this.sortOptionsByGroup(options),
-      clearable,
-      backspaceRemoves: t.Nil.is(backspaceRemoves)
-        ? clearable
-        : backspaceRemoves,
-      resetValue: null,
-      className: cx("dropdown", className, this.getCustomClassNames()),
-      value: this.valueToOption(this.props.value, options),
-      onInputKeyDown,
-      onChange: _onChange,
-      menuRenderer: this.menuRenderer as SelectNS.MenuRendererHandler
-    };
-
     return (
       <Select
-        {...selectProps}
+        {...props}
+        isClearable={isClearable}
```

Comment:
why is this explicit?

Reply:
leftover :)
it was used to calculate `backspaceRemovesValue`